### DISCUSSION TITLE
Toggle switch every 10 minutes between 12:00 and 18:00

### DISCUSSION THREAD

**Discussion by JsBergbau** [arendst/Tasmota#17942] [Q&A] [answered]
Tasmota / Gosund Plug should turn on for 10 minutes, then turn off for 10 minutes and so on between 12:00 and 18:00.

I think rules feature is therefore needed. https://tasmota.github.io/docs/Rules/#rule-syntax
However since conditional rules are not built into precompiled binaries I think there must be some other solution.
I found so far the PulseTime, but as far as I understand that is swichting of after a certain amount of time.

How can this task be achieved by an easy way?

**Comment by Noschvie** [comment, marked as answer]
Have a look to [Tasmota Shelly 1 Plus und Zeitsteuerung](https://forum.creationx.de/forum/index.php?thread/3746-tasmota-shelly-1-plus-und-zeitsteuerung/&postID=40661#post40661)

**Reply by barbudor** [reply]
The reflex to look for "Conditionnal Rules" often comes from previous experience in programming language
But rules are not a programming language
In most of the case, you do not need the IF/THEN/ELSE feature

In your exemple, considering that
- 12:00 is 720 minutes past midnight (12*60)
- 18:00 is 1080 minutes pas midnight
- You want 10 minutes ON then 10 minutes OFF which is a 20 minutes cycle

I would suggest the following:
```
PulseTime 700

Rule1
  on time#minute<720 do break
  on time#minute>=1080 do break
  on time#minute|20 do power1 on endon

Rule1 1
```

The ruleset "Rule1" must be entered all in the same line/command. I have written it on 3 lines so you can see clearly the 3 rules in the same ruleset
The 1st and 2nd rules will do nothing when the time is outside the range [10:00 - 18:00[
The 3rd rule will power on every 20 minutes
Because of `PulseTime 700`  the power will remains on only for 10 minutes
The 10 minutes off duration comes from 20 - 10

BREAK is used to stop the evaluation of the rules that are based on the **same** trigger (here `clock#time`)
So the order of the rules is important
It is more or less equivalent to
```
on time#minute do 
   if %value% < 720 do nothing
   else if %value% >= 1080 do nothing
   else if %value% modulo 20 min = 0 do power on
```

I haven't tested but I'm 95% confident that it will work

**Reply by sfromis** [reply]
When on and off periods are the same length, you could also use `Power toggle` at 10-minute intervals, thus also respecting any prior state of the relay output.

**Reply by barbudor** [reply]
With Power Toggle, you need to be sure to have an even count of operations or you will end-up with a Power ON
That won't happen with PulseTime

**Reply by sfromis** [reply]
That depends on the detail with respecting initial state, which may or may not be desirable. If so, it may indeed be relevant to turn off after period expired.

**Comment by JsBergbau** [comment]
@Noschvie Thanks for your solution.

I've tried with
```PulseTime 700

Rule1
  on time#minute<720 do break
  on time#minute>=1080 do break
  on time#minute|20 do power1 on endon

Rule1 1```
However switch stays off all the time. 
When switching on it goes off after 10 minutes due to pulsetime. That part works at least.
What did I miss?

**Reply by sfromis** [reply]
Check the log for what happens or not....

I did a small test shot which looks good:
```
on Time#Minute<1070 do break 
on Time#Minute>1080 do break 
on Time#Minute|2 do var1 %time% endon

17:49:00.704 RUL: TIME#MINUTE<1070 performs ""
17:50:00.706 RUL: TIME#MINUTE|2 performs "var1 1070"
17:52:00.702 RUL: TIME#MINUTE|2 performs "var1 1072"
17:54:00.727 RUL: TIME#MINUTE|2 performs "var1 1074"
17:56:00.709 RUL: TIME#MINUTE|2 performs "var1 1076"
17:58:00.707 RUL: TIME#MINUTE|2 performs "var1 1078"
18:00:00.714 RUL: TIME#MINUTE|2 performs "var1 1080"
```

**Comment by JsBergbau** [comment]
Thanks. Works now. Problem was time was not synced. 
Tasmota devices are in an environment without access to internet. With tasmota 8 time was fetched from DHCP Server. With version 12 time was not synced. After changing NTP to local NTP Server it worked. Has this changed in later versions than 8 that time from DHCP-Server is not used anymore?

**Reply by barbudor** [reply]
Most of home configuration do not provide NTP server in DHCP
So this has been abandonned
If you wat to use a local NTP server you have to specifiy yourself wiht `NtpServer` command, just like you did.

**Reply by sfromis** [reply]
And even in cases where some router was "trying" to announce NTP via DHCP, it often did not work, and led to confusing problems. Too much trouble for it to be a viable feature.

**Reply by JsBergbau** [reply]
It's not getting the NTP server via DHCP. In version 8 Tasmota tried to get the time via NTP from routers IP address and that worked perfectly. I've opened an issue about that problem since having this feature doesn't hurt, but saves a lot of trouble https://github.com/arendst/Tasmota/issues/17984

**Reply by sfromis** [reply]
No, old Tasmota did **not** just use the IP address of the router for NTP, but instead what was DHCP-announced by the router. And the badness of that caused a lot of trouble.

**Reply by JsBergbau** [reply]
The documentation said it tries to get time via NTP from the routers IP address. At least it worked for me and there is neither Option 042 nor Option 004 in the DHCP packets of my router.
![grafik](https://user-images.githubusercontent.com/37013344/219942570-2c2a4535-c70d-408c-92a4-7795cd984045.png)

Lets discuss this in that issue https://github.com/arendst/Tasmota/issues/17984 I think it fits better there.
